packages/@aws-cdk/aws-iam/lib/policy-statement.ts

```diff
@@ -408,6 +408,42 @@ export class PolicyStatement {
     }
     this.addConditions(conditions);
   }
+
+  /**
+   * Validate that the policy statement satisfies base requirements for a policy.
+   */
+  public validateForAnyPolicy(): string[] {
+    const errors = new Array<string>();
+    if (this.action.length === 0 && this.notAction.length === 0) {
+      errors.push('A PolicyStatement must specify at least one allow or deny action.');
```

Comment:
Yes, however, there is also `effect: Allow | Deny`, so I wonder if this might not be confusing what the actual check is.

`Allow NotActions` is not the same as `Deny Actions`.

```suggestion
      errors.push('A PolicyStatement must specify at least one \'action\' or \'notAction\'.');
```

Reply:
Good catch - I forgot about this distinction!